packages/itwinui-css/src/button/split-button.scss

```diff
@@ -3,43 +3,54 @@
 @import '../style/index';
 
 @mixin iui-button-split {
-  @include iui-button-group;
+  display: inline-flex;
+  align-items: center;
+  isolation: isolate;
 
-  &:hover {
-    background-color: rgba(var(--iui-color-foreground-body-rgb), var(--iui-opacity-6));
+  :where(.iui-button:first-child) {
+    border-top-right-radius: 0;
+    border-bottom-right-radius: 0;
+
+    &[data-iui-variant='high-visibility'],
+    &[data-iui-variant='cta'] {
+      border-right-color: rgba(var(--iui-color-foreground-accessory-rgb), var(--iui-opacity-4));
+    }
+
+    &[disabled],
+    &:disabled,
+    &[aria-disabled='true'] {
+      border-right-color: var(--iui-color-background-4);
+    }
+
+    &:where(:enabled:hover) {
```

Comment:
We should not be using `:enabled` as it won't work with the `[disabled]` or `aria-disabled` approaches.

Instead, this block should be moved up before the disabled block, and then we should set another z-index (maybe -1) for the disabled block.

Reply:
Oh, I hadn't thought that `:enabled` wouldn't effect `aria-disabled` or `[disabled]`.

I've made some changes that work with all 3 of the examples in the screenshot below (not saying they are valid cases, but the z-index is ordered correctly for all situations).
![Screen Shot 2022-07-27 at 12 12 57 PM](https://user-images.githubusercontent.com/849817/181297037-53ba0197-e6bb-4671-a66d-4feaf2bfbf8c.png)

Reply:
i think we don't need this extra rule
![image](https://user-images.githubusercontent.com/9084735/181308024-79ee2b82-65f1-4b04-a920-63f858d8a115.png)

any time we have to write `&, &:hover`, that feels like a mistake. Does adding `z-index: -1` (without `:hover`) after line 28 not work?

Reply:
The middle example has a z-index order issue, but that may be a non-issue since this button combo may be invalid.

![Screen Shot 2022-07-27 at 2 56 16 PM](https://user-images.githubusercontent.com/849817/181350605-2b47aa45-78ca-420a-8aee-5292412d82c9.png)

Reply:
Why not just apply z-index: -1 on the disabled button instead of unsetting it?
![ ](https://user-images.githubusercontent.com/9084735/183152028-186ee2c8-db13-4866-8124-0ea080a3d696.png)

Reply:
This suggestion works and keeps specificity down at 0, 2, 0!  Change made.